mantidproject/mantid
# IN5 mask non overlapping bins

**Description of work.**
See the issue description.



**To test:**

Make sure you have the system test data in your path. Run the script:
```
import numpy as np

def ecrase(ws):
    y = ws.extractY()
    y_prime = np.full_like(y,0.1)
    for i in range(ws.getNumberHistograms()):
        ws.setY(i, y_prime[i])

run = 'ILL/IN5/176053.nxs'

DirectILLCollectData(
    Run=run,
    Normalisation='Normalisation Time',
    OutputIncidentEnergyWorkspace='Ei',
    OutputElasticChannelWorkspace='Elc',
    OutputWorkspace='raw',
    OutputEPPWorkspace='Epp'
)

ws = Load(run)
ecrase(ws)

DirectILLCollectData(
    InputWorkspace='ws',
    Normalisation='Normalisation Time',
    IncidentEnergyWorkspace='Ei',
    ElasticChannelWorkspace='Elc',
    OutputWorkspace='Eone'
)

DirectILLReduction(
    InputWorkspace='Eone',
    EnergyRebinning='a',
    QBinningParams='',
    OutputWorkspace='red'
)

# - - - Energy scan - - - #
#k=0
Ei=  dt.SampleLogs(mtd['red']).Ei
ki= np.sqrt(Ei/2.072125902)
Q= 3 # ki
dQ= 0.5 # 0.2*Q
eMin=-36
eMax= 16# 0.92*Ei
yMin=0
yMax= 0.0001
fig, axes, cuts= dt.plotconstQ(['red'],Q,dQ) 
axes.set_xlim(xmin=eMin,xmax=eMax)
axes.set_ylim(ymin=yMin,ymax=yMax) 
axes.legend(**{'loc': 'upper left'})
#if saveFigs: fig.savefig(sampleLabels[k]+'E.png')
fig.show()
```
In the right end of the spectrum the curve should **NOT** drop down, as the green curve in the plot attached to the issue.



Fixes #27801 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
